job-server/src/spark.jobserver/JobManagerActor.scala

```diff
@@ -158,7 +158,7 @@ class JobManagerActor(contextConfig: Config) extends InstrumentedActor {
 
     case KillJob(jobId: String) => {
       jobContext.sparkContext.cancelJobGroup(jobId)
-      statusActor ! JobKilled(jobId, DateTime.now())
+      statusActor ! JobKilled(jobId, DateTime.now(), new Exception("killed"))
```

Comment:
Why don't we use a specific SJS exception instead of generic one with a message?  Matching on types is much safer than matching on strings .... though I agree the probability of a user doing this is pretty low.